client/components/AddSponsor.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+
+interface AddSponsorProps {
+  eventId: string;
+  chapterId: string;
+}
+
+interface SponsorData {
+  name?: string;
+  website?: string;
+  type?: string;
+}
+
+const AddSponsor: React.FC<AddSponsorProps> = ({ eventId, chapterId }) => {
+  const [isSubmitting, setSubmitting] = React.useState<boolean>(false);
+  const [values, setValues] = React.useState<SponsorData>({ type: 'FOOD' });
```

Comment:
for this i think it would be better to make the fields of `SponsorData` not optional, and instead pass in a default value of empty string.  this will let you do some validations on the values without having to check for null/undefined-ness.  e.g.

```
interface SponsorData {
  name: string;
  website: string;
  type: string;
}

...

React.useState<SponsorData>({ name: '', website: '', type: 'FOOD' })
```

Reply:
+1 on this

Reply:
true I didn't think of that, I kept them optional because they were initially not set, didn't think to set them to nothing haha. Have fixed.